File: Backend/SoftLibOrdenBO/src/pe/edu/pucp/softlib/orden/bo/OrdenAbastecimientoBO.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.edu.pucp.softlib.orden.bo;

import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.softlib.orden.dao.OrdenAbastecimientoDAO;
import pe.edu.pucp.softlib.orden.daoImp.OrdenAbastecimientoDAOImpl;
import pe.edu.pucp.softlib.orden.model.EstadoDeOrden;
import pe.edu.pucp.softlib.orden.model.LineaDeOrden;
import pe.edu.pucp.softlib.orden.model.OrdenAbastecimiento;
import pe.edu.pucp.softlib.usuario.model.Empleado;

/**
 *
 * @author Joshua Haro
 */
public class OrdenAbastecimientoBO {
    private final OrdenAbastecimientoDAO ordenAbastecimientoDAO;
    
    public OrdenAbastecimientoBO(){
        this.ordenAbastecimientoDAO = new OrdenAbastecimientoDAOImpl();
    }
    
    public Integer insertar(ArrayList<LineaDeOrden> lineasDeOrdenes,
            EstadoDeOrden estadoDeOrden, Date fechaCreacion, Double total, 
            Integer idEmpleado, Date fechaRecepcion, String descripcion) {
        Empleado empleado = new Empleado();
        empleado.setIdPersona(idEmpleado);
        OrdenAbastecimiento ordenAbastecimiento = 
                new OrdenAbastecimiento(null, lineasDeOrdenes, 
                estadoDeOrden, fechaCreacion, total, empleado,true,
                fechaRecepcion,descripcion);
        return this.ordenAbastecimientoDAO.insertar(ordenAbastecimiento);
    }
    
    public Integer modificar(Integer idOrdenAbastecimiento,
            ArrayList<LineaDeOrden> lineasDeOrdenes,
            EstadoDeOrden estadoDeOrden, Date fechaCreacion, Double total, 
            Integer idEmpleado,Boolean activo, Date fechaRecepcion, String descripcion) {
        Empleado empleado = new Empleado();
        empleado.setIdPersona(idEmpleado);
        OrdenAbastecimiento ordenAbastecimiento = 
                new OrdenAbastecimiento(idOrdenAbastecimiento, lineasDeOrdenes, 
                estadoDeOrden, fechaCreacion, total, empleado, activo,
                fechaRecepcion,descripcion);
        return this.ordenAbastecimientoDAO.modificar(ordenAbastecimiento);
    }
    
    public Integer eliminar(Integer idOrdenAbastecimiento) {
        OrdenAbastecimiento ordenAbastecimiento = new OrdenAbastecimiento();
        ordenAbastecimiento.setIdOrden(idOrdenAbastecimiento);
        return this.ordenAbastecimientoDAO.eliminar(ordenAbastecimiento);
    }
    
    public ArrayList<OrdenAbastecimiento> listarTodos(){
        return this.ordenAbastecimientoDAO.listarTodos();
    }
    
    public OrdenAbastecimiento obtenerPorId(Integer idOrdenAbastecimiento){
        return this.ordenAbastecimientoDAO.obtenerPorId(idOrdenAbastecimiento);
    }
    
    public Boolean existeOrdenAbastecimiento(Integer idOrdenAbastecimiento){
        OrdenAbastecimiento ordenAbastecimiento = new OrdenAbastecimiento();
        ordenAbastecimiento.setIdOrden(idOrdenAbastecimiento);
        return this.ordenAbastecimientoDAO.existeOrdenAbastecimiento
                    (ordenAbastecimiento);
    }    
}

File: Frontend/SoftLibOrdenBO/OrdenAbastecimientoBO.cs
using SoftLibBO.ServicioWeb;
using SoftLibBO;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftLibOrdenBO
{
    public class OrdenAbastecimientoBO : BaseBO
    {
        public int insertar(BindingList<lineaDeOrden> lineasDeOrdenes,
            estadoDeOrden estadoDeOrden, DateTime fechaCreacion, double total,
            int idEmpleado, DateTime fechaRecepcion, string descripcion)
        {
            lineaDeOrden[] lineas = lineasDeOrdenes.ToArray();
            return this.WsOrdenAbastecimiento.ordenAbastecimiento_insertar(lineas,
             estadoDeOrden, fechaCreacion, total,
             idEmpleado, fechaRecepcion, descripcion);
        }

        public int modificar(int idOrdenAbastecimiento,
               BindingList<lineaDeOrden> lineasDeOrdenes,
                estadoDeOrden estadoDeOrden, DateTime fechaCreacion, double total,
                int idEmpleado, Boolean activo, DateTime fechaRecepcion, string descripcion)
        {
            lineaDeOrden[] lineas = lineasDeOrdenes.ToArray();
            return this.WsOrdenAbastecimiento.ordenAbastecimiento_modificar(idOrdenAbastecimiento,
                lineas, estadoDeOrden, fechaCreacion, total,
                 idEmpleado, activo, fechaRecepcion, descripcion);
        }

        public int eliminar(int idOrdenAbastecimiento)
        {
            return this.WsOrdenAbastecimiento.ordenAbastecimiento_eliminar(idOrdenAbastecimiento);
        }

        public BindingList<ordenAbastecimiento> listarTodos()
        {
            ordenAbastecimiento[] arreglo = this.WsOrdenAbastecimiento.ordenAbastecimiento_listarTodos();
            return new BindingList<ordenAbastecimiento>(arreglo);
        }

        public ordenAbastecimiento obtenerPorId(int idOrdenAbastecimiento)
        {
            return this.WsOrdenAbastecimiento.ordenAbastecimiento_obtenerPorId(idOrdenAbastecimiento);
        }

        public Boolean existeOrdenAbastecimiento(int idOrdenAbastecimiento)
        {
            return this.WsOrdenAbastecimiento.existeOrdenAbastecimiento(idOrdenAbastecimiento);
        }
    }
}
